hazelcast/src/main/java/com/hazelcast/jet/core/processor/SinkProcessors.java

```diff
@@ -397,7 +399,8 @@ public static <T> ProcessorMetaSupplier writeJdbcP(
         checkNotNull(dataSourceSupplier, "dataSourceSupplier");
         checkNotNull(bindFn, "bindFn");
         checkPositive(batchLimit, "batchLimit");
-        return WriteJdbcP.metaSupplier(jdbcUrl, updateQuery, ctx -> dataSourceSupplier.get(), bindFn, exactlyOnce, batchLimit);
+        return WriteJdbcP.metaSupplier(jdbcUrl, updateQuery, ctx -> DataStoreHolder.closing(dataSourceSupplier.get()),
```

Comment:
Maybe we shouldn't close data-source when we don't create it? 

It seems to me, if we also do this, we can completely get rid of `DataStoreHolder` and rely on `ExternalDataStoreFactory#close` when closing is necessary.

Reply:
> Maybe we shouldn't close data-source when we don't create it?

Yes, I think it makes sense. I've changed it to nonClosing holder

> 
> It seems to me, if we also do this, we can completely get rid of `DataStoreHolder` and rely on `ExternalDataStoreFactory#close` when closing is necessary.

`ExternalDataStoreFactory` is closed during the shutdown of the instance. We would end up with multiple non-shared datastores eating resources without any need until the shutdown